File: doc-examples/hibernate6-example-java/src/main/java/example/MonthDayDateAttributeConverter.java
package example;

import jakarta.persistence.AttributeConverter;
import jakarta.persistence.Converter;
import java.time.LocalDate;
import java.time.MonthDay;
import java.sql.Date;

@Converter
public class MonthDayDateAttributeConverter implements AttributeConverter<MonthDay, Date> {
    @Override
    public Date convertToDatabaseColumn(MonthDay monthDay) {
        if (monthDay == null) {
            return null;
        }
        return Date.valueOf(monthDay.atYear(2000));
    }

    @Override
    public MonthDay convertToEntityAttribute(Date date) {
        if (date == null) {
            return null;
        }
        LocalDate localDate = date.toLocalDate();
        return MonthDay.of(
                localDate.getMonth(),
                localDate.getDayOfMonth()
        );
    }
}

File: doc-examples/hibernate-example-python/src/main/python/example/MonthDayDateAttributeConverter.py
from datetime import date

from jakarta.persistence import AttributeConverter, Converter
from java.sql import Date
from java.time import MonthDay


@Converter
class MonthDayDateAttributeConverter(AttributeConverter[MonthDay, Date]):

    def convertToDatabaseColumn(self, month_day: MonthDay | None) -> Date | None:
        if month_day is None:
            return None
        return Date.valueOf(month_day.atYear(2000))

    def convertToEntityAttribute(self, value: Date | None) -> MonthDay | None:
        if value is None:
            return None
        local_date = value.toLocalDate()
        return MonthDay.of(local_date.getMonth(), local_date.getDayOfMonth())
